File: go/gin/internal/modules/address/module.go
package address

import (
	auth "tcc/internal/modules/auth/middleware"
	"tcc/internal/modules/client"

	"github.com/gin-gonic/gin"
)

func AddressModule(router *gin.Engine) {
	service := AddressService{
		repository: struct{client client.ClientRepository; address AddressRepository}{
			client: client.ClientRepository{},
			address: AddressRepository{},
		},
	}
	controller := AddressController{
		service: service,
	}

	group := router.Group("/enderecos")
	{
		group.POST("", auth.AuthMiddleware(),controller.Create)
	}
}

File: rust/actix/src/modules/address/module.rs
use actix_web::{web, Scope, dev::ServiceFactory};
use actix_web::dev::ServiceRequest;
use actix_web::dev::ServiceResponse;
use actix_web::Error;
use super::controller::AddressController;
use crate::modules::auth::middleware::AuthMiddleware;

pub fn address_module() -> Scope<impl ServiceFactory<ServiceRequest, Config = (), Response = ServiceResponse, Error = Error, InitError = ()>> {
    web::scope("/enderecos")
        .wrap(AuthMiddleware)
        .route("", web::post().to(AddressController::create))
}